### PR TITLE
Protected attributes

### PR DESCRIPTION
**Pull request by peiyaoh**
### Description

- Support Get/Set/Delete protected attributes stored in a separate store to provide a more restricted approach to manage sensitive data and reduce accidental leaking. The methods are exposed to 3RD devs through UserManager.

See the proposal: https://umich-mia.atlassian.net/wiki/spaces/U/pages/191168513/JustIn+Protected+Attributes

### Related Jira Tickets

- https://umich-mia.atlassian.net/browse/JST-112?atlOrigin=eyJpIjoiZWUzZDFiMmJhMjU2NDE0ZWFiYTY5MGQ4NjIwYzk5MGYiLCJwIjoiaiJ9

Please ensure all items are complete before requesting a review:

- [x] **Code Completed**: All code for the feature or bugfix has been implemented.
- [x] **Unit Tests Added**: Unit tests are written for all new functionality, covering edge cases where applicable.
- [ ] **Integration Tests (where necessary)**: Integration tests are added for interactions between modules or components.
- [x] **All Tests Passing**: All unit and integration tests are passing.
- [x] **Type Safety Checked**: Ensure TypeScript types are correctly defined and used. Avoid `any` types without proper justification.
- [x] **Typedoc Comments Added**: Ensure all functions, classes, interfaces, and types are documented using TypeDoc syntax.
- [x] **Code Formatting**: All code files are formatted according to the project’s standards (e.g., Prettier, ESLint) and no linting errors remain.
- [x] **No Console Logs**: Remove any `console.log` or other debugging output from the code.
- [ ] **Documentation Updated**: Update relevant project documentation (e.g., README, configuration guides) if necessary.

---

### CODE REVIEW

File: src/protected-manager/protected-manager.ts [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+ * Initializes the ProtectedManager by initializing the DataManager and creating indexes.
+ *
+ * @returns {Promise<void>} Resolves when initialization is complete.
+ */
+const init = async (): Promise<void> => {
+  await dm.init();
+  await dm.ensureStore(PROTECTED);
+  await dm.ensureIndexes(PROTECTED, [
+    {
+      name: 'uniq_user_identifier_namespace',
```
Comment (jadrubio) [inline src/protected-manager/protected-manager.ts]: Do we want to move this to the top as a constant?

**Comment by peiyaoh** [comment]
Coppied to justin-consotium. Updates will be done in the new branch.
See https://github.com/justin-consortium/justin-core/tree/add-protected-attributes-draft
